core/src/main/java/org/mule/runtime/core/el/LifecycleAwareBindingContext.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.el;
+
+import static java.util.stream.Collectors.toList;
+import static org.mule.runtime.core.api.lifecycle.LifecycleUtils.initialiseIfNeeded;
+import static org.mule.runtime.core.api.lifecycle.LifecycleUtils.startIfNeeded;
+import static org.mule.runtime.core.api.lifecycle.LifecycleUtils.stopIfNeeded;
+import org.mule.runtime.api.el.Binding;
+import org.mule.runtime.api.el.BindingContext;
+import org.mule.runtime.api.el.ExpressionModule;
+import org.mule.runtime.api.exception.MuleException;
+import org.mule.runtime.api.lifecycle.Initialisable;
+import org.mule.runtime.api.lifecycle.InitialisationException;
+import org.mule.runtime.api.lifecycle.Startable;
+import org.mule.runtime.api.lifecycle.Stoppable;
+import org.mule.runtime.api.metadata.TypedValue;
+
+import java.util.Collection;
+import java.util.Optional;
+
+/**
+ * Implementation of {@link BindingContext} that supports lifecycle execution on the elements of the context.
+ * @since 4.0
+ */
+public class LifecycleAwareBindingContext implements BindingContext, Initialisable, Startable, Stoppable {
```

Comment:
I wouldn't do this. I think it's better to pull the activator concept from the `optimus` branch and have this done on the lifecycle of the ExtensionManager